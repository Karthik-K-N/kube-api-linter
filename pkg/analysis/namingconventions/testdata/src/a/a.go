package a

// Shouldn't care about Go types
type BasketOfFruit struct {
	RedFruit    []string `json:"redFruit,omitempty"`    // want `naming convention "nofruit": field RedFruit: no fields should contain any variations of the word 'fruit' in their name`
	OrangeFruit []string `json:"orangeFruit,omitempty"` // want `naming convention "nofruit": field OrangeFruit: no fields should contain any variations of the word 'fruit' in their name`
	GreenFruit  []string `json:"greenFruit,omitempty"`  // want `naming convention "nofruit": field GreenFruit: no fields should contain any variations of the word 'fruit' in their name`
	FruitBlue   []string `json:"fruitBlue,omitempty"`   // want `naming convention "nofruit": field FruitBlue: no fields should contain any variations of the word 'fruit' in their name`
	Fruit       []string `json:"fruit,omitempty"`       // want `naming convention "nofruit": field Fruit: no fields should contain any variations of the word 'fruit' in their name`
	AFruit      string   `json:"aFruit,omitempty"`      // want `naming convention "nofruit": field AFruit: no fields should contain any variations of the word 'fruit' in their name`
}

// Shouldn't care about methods
func (b BasketOfFruit) GrabFruit() string {
	return "nothing"
}

// shouldn't care about functions
func IsFruit(in string) {
	return
}

type SpecialBehaviors struct {
	SomethingBehavior string `json:"somethingBehavior,omitempty"` // want `naming convention "preferbehaviour": field SomethingBehavior: prefer the use of the word 'behaviour' instead of 'behavior'.`
	BehaviorCrazy     bool   `json:"behaviorCrazy,omitempty"`     // want `naming convention "preferbehaviour": field BehaviorCrazy: prefer the use of the word 'behaviour' instead of 'behavior'.`
}

type Configurations struct {
	SomeSupportedThingy string `json:"someSupportedThingy,omitempty"`
	UnsupportedThingy   string `json:"unsupportedThingy,omitempty"` // want `naming convention "nounsupported": field UnsupportedThingy: no fields allowing for unsupported behaviors allowed`
}

type TestSet struct {
	TestName string `json:"testName,omitempty"`  // want `naming convention "notest": field TestName: no temporary test fields`
	Other    string `json:"otherTest,omitempty"` // want `naming convention "notest": field Other: no temporary test fields`
}
