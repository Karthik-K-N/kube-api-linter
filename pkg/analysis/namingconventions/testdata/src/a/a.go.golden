package a

// Shouldn't care about Go types
type BasketOfFruit struct {
	Red    []string `json:"red,omitempty"`    // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
	Orange []string `json:"orange,omitempty"` // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
	Green  []string `json:"green,omitempty"`  // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
	Blue   []string `json:"blue,omitempty"`   // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
	Fruit  []string `json:"fruit,omitempty"`  // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
	A      string   `json:"a,omitempty"`      // want `naming convention "nofruit": no fields should contain any variations of the word 'fruit' in their name`
}

// Shouldn't care about methods
func (b BasketOfFruit) GrabFruit() string {
	return "nothing"
}

// shouldn't care about functions
func IsFruit(in string) {
	return
}

type SpecialBehaviors struct {
	SomethingBehaviour string `json:"somethingBehaviour,omitempty"` // want `naming convention "preferbehaviour": prefer the use of the word 'behaviour' instead of 'behavior'.`
	BehaviourCrazy     bool   `json:"behaviourCrazy,omitempty"`     // want `naming convention "preferbehaviour": prefer the use of the word 'behaviour' instead of 'behavior'.`
}

type Configurations struct {
	SomeSupportedThingy string `json:"someSupportedThingy,omitempty"`
	// want `naming convention "nounsupported": no fields allowing for unsupported behaviors allowed`
}

type TestSet struct {
    TestName string `json:"testName,omitempty"` // want `naming convention "notest": no temporary test fields`
	Other    string `json:"otherTest,omitempty"` // want `naming convention "notest": no temporary test fields`
}
